### PR TITLE
adding in try catch for json parsing error

### PR DESCRIPTION
**Pull request by jtaylor-orbis**
### TL;DR
Added error handling for JSON parsing in the Catalyst query consumer and updated gitignore patterns for fly.toml files.

### What changed?
- Added try-catch block around JSON parsing in the consumer adapter
- Added detailed error logging when JSON parsing fails, including the query, variables, and raw response
- Updated .gitignore to handle different variations of fly.toml files
- Added fallback return of empty data object when parsing fails

### How to test?
1. Configure an invalid or malformed Catalyst query
2. Run the consumer
3. Verify that the error is properly logged with query details
4. Confirm the system continues to operate with the fallback empty data response

### Why make this change?
To improve system resilience by gracefully handling JSON parsing failures in the Catalyst query consumer. This prevents the application from crashing when receiving malformed responses and provides better debugging information when issues occur.

### CODE REVIEW

**Comment by jtaylor-orbis** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/38?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#38** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/38?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#37** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/37?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#36** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/36?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/35?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#34** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/34?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#33** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/33?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#32** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/32?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#31** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/31?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#30** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/30?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#29** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/29?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#28** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/28?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#27** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/27?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/26?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/25?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/24?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#22** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/22?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21** <a href="https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/21?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jtaylor-orbis and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .gitignore [hunk 1]
```diff
@@ -182,8 +182,8 @@ adapter-db/
 yourdb/
 ./**/*.mdb
 config.docker.toml
-config.fly.toml
+config.fly.toml.bk
 db/*/**
 
 .tak_downloads/** 
-
+config.fly.toml*
```
Comment (cgmckeever-orbis) [inline .gitignore:189]: this is probably better off as `config.*` but this works for now

File: src/adapters/consumer/index.ts [hunk 2]
```diff
@@ -78,7 +78,15 @@ export class Consumer {
                 variables: new Object(this.config.consumer!.catalyst_query_variables)
             }),
         })
-        return  await result.json()
+        try {
+            return await result.json()
+        } catch (error) {
+            console.error("Error parsing response", error)
+            console.error("query", this.config.consumer!.catalyst_query)
```
Comment (jaeyojae) [inline src/adapters/consumer/index.ts:85]: adding a space ": " might improve readability but lgtm

**Comment by jtaylor-orbis** [comment]
### Merge activity

* **Nov 14, 1:03 PM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/orbisoperations/catalyst-tak-adapter/38).

**Pull request merged**